### ISSUE TITLE
Kafka/SASL password only configuration doesn't work

### ISSUE THREAD

**Issue by cbuckley01** [argoproj/argo-events#1217]
Config validation for Kafka/SAML is broken, when configuring just a password the contoller throws the following error:

`{"level":"error","ts":1621011775.476398,"logger":"argo-events.eventsource-controller","caller":"eventsource/controller.go:81","msg":"validation error","namespace":"argo-events","eventSource":"kafka","error":"invalid sasl config, please configure either User, and/or Password"`

This config works:
```      
     sasl:
        mechanism: PLAIN
        password:
          key: password
          name: my-user
        user:
          key: user
          name: my-user```

This will not:
```      
     sasl:
        mechanism: PLAIN
        password:
          key: password
          name: my-user```

**Comment by whynowy** [comment]
Forgive my poor Kafka knowledge, is password only config allowed? @joshuajorel @cbuckley01

**Comment by cbuckley01** [comment]
Right, that way be true, sorry, I might have been a bit confused by the error message

**Comment by joshuajorel** [comment]
@whynowy I'm not aware that you should be able to configure password only since the user value is usually used for ACL. I think he's referring to if you can reference the same field for the secret to contain the username and password. Should this be the implementation? 

@cbuckley01 User and Password values should be in two separate fields in the Secret object.

**Comment by cbuckley01** [comment]
Right, I have re-titled this issue with the root issue.  I will hopefully get you guys a PR today making things a bit more clear/consistent with TLS wording and add an examples to help clarify this for future users.

**Comment by whynowy** [comment]
https://github.com/argoproj/argo-events/pull/1220
